docs/Makefile

```diff
@@ -8,7 +8,7 @@ SPHINXBUILD   ?= sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = _build
 
-.PHONY: Makefile help clean rust-html
+.PHONY: Makefile help rust-html
```

Comment:
I mistakenly added this here which prevent `clean` from matching the `%` rule below. It instead matches an empty rule which does nothing. `sphinx-build -M clean ...`, on the other hand, deleted `docs/_build`.

This is rather important because Sphinx doesn't seem to invalidate its cache when the Python dylib/so changes.